README.rst

```diff
@@ -38,6 +39,25 @@ After installing this module, add the following to your `conf.py` to enable it
         'sphinxcontrib.prettyspecialmethods',
     ]
 
+Changing "self"
+---------------
+
+If a `meta info field`_ named :code:`selfref` is included in the docstring, its
+value will be used in place of "self" in the output:
+
+.. _meta info field: https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#info-field-lists
+
+.. code-block:: rst
+
+    .. method:: __add__(other)
+        Docstring
+        :meta selfref: obj
```

Comment:
How about `:meta self-param: obj`, which is remiscent of `:param: obj`? Then use `self_param` in the python code.

The source of `inspect.py` uses `self_parameter`, so I think `self_param` is a sensible choice.

Reply:
@eric-wieser I agree that "self param" is a better name. I'm using it in the revision I'm making.